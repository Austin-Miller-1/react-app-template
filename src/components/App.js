import React, {Component} from 'react';

/**
 * The top-level React component which hosts the entire React application.
 */
class App extends Component{
  constructor(){
    super();
  }

  render(){
    return (
      <React.Fragment>
       <h1>Application</h1>
       <p>This is some sample text!</p>
      </React.Fragment>
    );
  }
}

export {
  App as default,
};