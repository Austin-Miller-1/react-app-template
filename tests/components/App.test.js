import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from '../../src/components/App';
import { render } from '@testing-library/react';

describe('<App>', () => {
  it('renders a header', () => {
    const root = render(<App/>);
    expect(root.getByRole('heading').textContent).toBeTruthy();
  });

  it('renders some additional text on the page', () => {
    const root = render(<App/>);
    expect(root.getByText('This is some sample text!')).toBeTruthy();
  });

  describe('Snapshot tests', () => {
    it('matches the existing snapshot', () => {
      const actualSnapshot = TestRenderer
        .create(<App/>)
        .toJSON();
      expect(actualSnapshot).toMatchSnapshot();
    });
  });
});