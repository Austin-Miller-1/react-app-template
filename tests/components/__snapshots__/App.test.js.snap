// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> Snapshot tests matches the existing snapshot 1`] = `
Array [
  <h1>
    Application
  </h1>,
  <p>
    This is some sample text!
  </p>,
]
`;
